client/src/Components/Counter/OfficeDesk.tsx

```diff
@@ -1,52 +1,69 @@
-                            When the customer arrives, please click on the button below to proceed.
-                        </p>
+                    <Alert variant="success" >
+                        {service_type && ticket_number ? (
+                            <>
+                                <Alert.Heading className="large-heading">Next Customer is {ticket_number} for service S-{service_type}</Alert.Heading>
+                                <p>
+                                    When the customer arrives, please click on the button below to proceed.
+                                </p>
+                            </>
+                        ) : (
+                            <Alert.Heading className="large-heading">There is no customer to serve</Alert.Heading>
+                        )}
                         <Button onClick={handleClose} variant="outline-success">
                             Close
                         </Button>
                     </Alert>
-            ) : (
-                <>
-                    {ticket_number && <h2>Current Customer: {ticket_number}</h2>}
-                    <div style={{ display: 'flex', flexWrap: 'wrap', gap: '3rem', justifyContent: 'space-around' }}>
-                        <Button variant="primary" onClick={async () => {
-                            try {
-                               const response = "6";  //await API.getNextCustomer();
-                                if (response /*&& response.ticket_number*/) {
-                                    // Assuming you have a way to update the ticket_number in your component state
-                                    // You might need to lift the state up or use a context
-                                    setTicketNumber(response/*.ticket_number*/);
-                                    handleShow();
+                ) : (
+                    <>
+                        {ticket_number && <h2>Current Customer: {ticket_number}</h2>}
+                        <div style={{ display: 'flex', flexWrap: 'wrap', gap: '3rem', justifyContent: 'space-around' }}>
+                            <Button variant="primary" onClick={async () => {
+                                try {
+                                    const response = await API.getNextCustomer();
+                                    console.log(response);
+                                    if (response) {
+                                        if(response.ticketNumber==0){
+                                            setTicketNumber(undefined);
+                                            setServiceType(undefined);
+                                            handleShow();
+                                        }else{
+                                            setTicketNumber(response.ticketNumber);
+                                            setServiceType(response.serviceType);
+                                        }
+                                    }
```

Comment:
Just for more readability and clearness, it's better to move the onclick method above the code.

Reply:
I think that the code is inside the onClick... Tell me if i'm wrong